### DISCUSSION TITLE
NTRIP Server follow survey-in status

### DISCUSSION THREAD

**Discussion by ereditsdani** [semuconsulting/pygnssutils#54] [Q&A] [answered]
Hi!

I am using pygnssutils to start my NTRIP server from CLI. However I couldn't find a way to keep track of the status of the survey-in of the base station. I have seen that in PyGPSClient there is a way to keep track of that, and I wonder that is there any way to keep track of the survey-in status in pygnssutils as well?

Thanks you for your help!

**Comment by semuadmin** [comment, marked as answer]
Hi @ereditsdani 

The pygnssutils gnssserver utility running in NTRIP Caster mode doesn't actually set or monitor the Survey-In (SVIN) status - it assumes the required base station mode (FIXED or SURVEY-IN) has already been successfully configured.

If you need guidance on how to configure the receiver initially, have a look at this example from the pyubx2 library:

https://github.com/semuconsulting/pyubx2/blob/master/examples/f9p_basestation.py

If you want to monitor the SURVEY-IN status while it's in progress, the relevant data is available via the UBX NAV-SVIN message  - specifically the `dur`, `valid` and `active` attributes.  You could, for example, use the pygnssutils gnssdump utililty to monitor SVIN status while the survey is in progress - e.g. something like this:

```shell
gnssdump --port /dev/ttyACM0 --baudrate 38400 --timeout 5 --quitonerror 1 --protfilter 2 --msgfilter NAV-SVIN --outputhandler "lambda msg: print(f'survey duration: {msg.dur}, valid?: {msg.valid}, active? {msg.active}')"
```

At the start, `dur` = 0, `valid` = 0 and `active`=1. Once running, you should see the `dur` value steadily increasing and, once successful, the `valid` flag will change to '1' and the `active` flag will change to '0'.

Alternatively, you could use something like the following Python code snippet (this is essentially what happens in PyGPSClient):

```python
from serial import Serial
from pyubx2 import UBXReader

SURVEY_DURATION = 60  # required survey duration in seconds

stream = Serial('/dev/ttyACM0', 38400, timeout=5)
ubr = UBXReader(stream)
for raw, parsed in ubr:
   if parsed.identity == "NAV-SVIN":
      print(f"SVIN duration {parsed.dur} ({parsed.dur * 100/SURVEY_DURATION}%), valid? {parsed.valid}, active? {parsed.active}")
```

Hope this helps

**Reply by ereditsdani** [reply]
Thank you for your help, i have used the python code, and it worked like a charm! :)
